src/DatePicker.elm

```diff
@@ -412,9 +419,24 @@ datePicker { today, currentMonth, currentDates, pickedDate, settings } =
         onPicker ev =
             Json.succeed
                 >> onWithOptions ev
-                    { preventDefault = True
+                    { preventDefault = False
                     , stopPropagation = True
                     }
+
+        onChange handler =
+            on "change" <| Json.map handler <| Json.at [ "target", "value" ] Json.string
+
+        isCurrentYear selectedYear =
+            year currentMonth == selectedYear
+
+        durationOption selectedYear =
+            ( toString selectedYear
+            , option [ value (toString selectedYear), selected (isCurrentYear selectedYear) ]
+                [ text (toString <| selectedYear) ]
+            )
+
+        dropdownYear =
+            Html.Keyed.node "select" [ onChange ChangeYear, class "year-list" ] (List.map durationOption (yearRange currentMonth settings.changeYear))
```

Comment:
I'm getting a peculiar behavior locally. I don't know if it's `Keyed` related, since I haven't used it before, but it seems like it might be.

Here are some repro steps; can you get this locally?
1. Create an example using `MoreOrLess 5`
2. Open picker
3. Select the earliest possible year (2012)
4. The select now shows "2016".

What's odd is that the debugger shows what looks like correct state behavior: 2012 is the year of `currentMonth`. Does this behavior look familiar? Hoping that you know about any gotchas with `Keyed`, or that I'm barking up the wrong tree entirely :).